test/e2e/e2e_test.go

```diff
@@ -259,6 +259,123 @@ var _ = ginkgo.Describe("Kueue", func() {
+				}).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).Should(gomega.Succeed())
+			localQueue = testing.MakeLocalQueue("main", ns.Name).ClusterQueue("cluster-queue").Obj()
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).Should(gomega.Succeed())
+		})
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteLocalQueue(ctx, k8sClient, localQueue)).Should(gomega.Succeed())
+			gomega.Expect(util.DeleteAllJobsInNamespace(ctx, k8sClient, ns)).Should(gomega.Succeed())
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueue, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, onDemandRF, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, spotRF, true)
+		})
+
+		ginkgo.It("Should evict a running job from the queue after QueueingPolicy is changed to Never and suspend to true", func() {
+			sampleJob = (&testingjob.JobWrapper{Job: *sampleJob}).Image("gcr.io/k8s-staging-perf-tests/sleep:v0.0.3", []string{"50s"}).Obj()
+			lookupKey1 := types.NamespacedName{Name: sampleJob.Name, Namespace: sampleJob.Namespace}
+			//createdJob1 := &batchv1.Job{}
+			wll := &kueue.Workload{}
+
+			ginkgo.By("checking the job starts")
+			gomega.Expect(k8sClient.Create(ctx, sampleJob)).Should(gomega.Succeed())
+
+			lookupKey := types.NamespacedName{Name: sampleJob.Name, Namespace: sampleJob.Namespace}
+			createdJob := &batchv1.Job{}
+			gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob)).Should(gomega.Succeed())
+
+			gomega.Eventually(func() *bool {
+				gomega.Expect(k8sClient.Get(ctx, lookupKey1, sampleJob)).Should(gomega.Succeed())
+				return sampleJob.Spec.Suspend
+			}, util.Timeout, util.Interval).Should(gomega.Equal(ptr.To(false)))
+
+			wlKey := types.NamespacedName{Name: workloadjob.GetWorkloadNameForJob(sampleJob.Name), Namespace: sampleJob.Namespace}
+
+			gomega.Eventually(func() bool {
+				if err := k8sClient.Get(ctx, wlKey, wll); err != nil {
+					return false
+				}
+				return workload.HasQuotaReservation(wll)
+			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+
+			ginkgo.By("Change the QueueingPolicy, suspend job and check the job remains suspended and the workload evicted")
+			// Changing QueuingPolicy to Never
+			wll.Spec.QueueingPolicy = kueue.QueueingPolicyTypeNever
+			gomega.Expect(k8sClient.Update(ctx, wll)).Should(gomega.Succeed())
+
+			// Suspending the job
+			gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob)).Should(gomega.Succeed())
+			createdJob.Spec.Suspend = ptr.To(true)
+			gomega.Expect(k8sClient.Update(ctx, createdJob)).Should(gomega.Succeed())
+
+			// Checking job stays suspended
+			ginkgo.By("checking job stays suspended")
+			gomega.Eventually(func() *bool {
+				gomega.Expect(k8sClient.Get(ctx, types.NamespacedName{Name: sampleJob.Name, Namespace: sampleJob.Namespace}, createdJob)).
+					Should(gomega.Succeed())
+				return createdJob.Spec.Suspend
+			}, util.Timeout, util.Interval).Should(gomega.Equal(ptr.To(true)))
```

Comment:
Not a very useful check.

Better to add a check below where you use `Consistently` to also check that the job is suspended as well.

See this example on how we check multiple things within one `Eventually` https://github.com/kubernetes-sigs/kueue/blob/337d7c9dfe16fb358bc032f20564a421c8aa3792/test/e2e/e2e_test.go#L239